main.tf

```diff
@@ -0,0 +1,74 @@
+data "aws_iam_policy_document" "lambda_trust_policy" {
+  statement {
+    actions = ["sts:AssumeRole"]
+    effect  = "Allow"
+
+    principals {
+      type        = "Service"
+      identifiers = ["lambda.amazonaws.com"]
+    }
+  }
+}
+
+resource "aws_iam_role" "lambda" {
+  name               = "${var.lambda_name}"
+  assume_role_policy = "${data.aws_iam_policy_document.lambda_trust_policy.json}"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_basic_role" {
+  role       = "${aws_iam_role.lambda.name}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"
+}
+
+resource "aws_iam_role_policy_attachment" "lambda_eni_management" {
+  role       = "${aws_iam_role.lambda.name}"
+  policy_arn = "arn:aws:iam::aws:policy/service-role/AWSLambdaENIManagementAccess"
+}
+
+resource "aws_iam_role_policy" "lambda" {
+  name   = "${var.lambda_name}"
+  role   = "${aws_iam_role.lambda.name}"
+  policy = "${var.iam_policy_document}"
+  count  = "${var.require_additional_policy}"
```

Comment:
**require_additional_policy** is boolean, why is it set directly into count?

Reply:
I'll remove this. Let me update this.